lib/std/fmt.zig

```diff
@@ -1454,6 +1458,8 @@ fn parseWithSign(
 
     var x: T = 0;
 
+    if (buf_start[0] == '_' or buf_start[buf_start.len - 1] == '_') return error.InvalidCharacter;
```

Comment:
Needs a check on `buf_start` length

Reply:
https://github.com/ziglang/zig/blob/d10ec6e70dbcf49ba14be549069271e564fec76b/lib/std/fmt.zig#L1423
^^ Does this not do it?

Reply:
Not if you have a prefix, but the check below should take care of this case so carry on.